ibc_middleware.go

```
@@ -0,0 +1,117 @@
+
+	return i.ics4Wrapper.SendPacket(ctx, chanCap, sourcePort, sourceChannel, timeoutHeight, timeoutTimestamp, data)
+}
+
+// WriteAcknowledgement implements the ICS4 Wrapper interface.
+func (i IBCMiddleware) WriteAcknowledgement(ctx sdk.Context, chanCap *capabilitytypes.Capability, packet exported.PacketI, ack exported.Acknowledgement) error {
+	return i.ics4Wrapper.WriteAcknowledgement(ctx, chanCap, packet, ack)
+}
+
+// GetAppVersion implements the ICS4 Wrapper interface.
+func (i IBCMiddleware) GetAppVersion(ctx sdk.Context, portID, channelID string) (string, bool) {
+	return i.GetAppVersion(ctx, portID, channelID)
```

golangci-lint (1.62.2)

SA5007: infinite recursive call

(staticcheck)